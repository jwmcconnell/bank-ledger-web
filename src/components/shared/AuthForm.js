import React from 'react';
import PropTypes from 'prop-types';

const styles = {
  form: {
    display: 'flex',
    flexDirection: 'column',
    width: 400,
    alignItems: 'center',
    margin: 'auto'
  },
  label: {
    margin: 10
  }
};

const AuthForm = ({ handleSubmit, handleUpdate, errorMessage }) => {
  return (
    <form onSubmit={handleSubmit} style={styles.form}>
      <label style={styles.label}>Username:
        <input name="username" type="test" onChange={handleUpdate}></input>
      </label>
      <label style={styles.label}>Password:
        <input name="password" type="password" onChange={handleUpdate}></input>
      </label>
      <button>Submit</button>
      <span>{errorMessage}</span>
    </form>
  );
};

AuthForm.propTypes = {
  handleUpdate: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  errorMessage: PropTypes.string
};

export default AuthForm;
