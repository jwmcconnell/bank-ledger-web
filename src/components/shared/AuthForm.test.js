import React from 'react';
import { shallow } from 'enzyme';
import AuthForm from './AuthForm';

describe('AuthForm component', () => {
  it('renders AuthForm', () => {
    const wrapper = shallow(<AuthForm 
      handleSubmit={() => {}} 
      handleUpdate={() => {}} 
      errorMessage={'error'}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
