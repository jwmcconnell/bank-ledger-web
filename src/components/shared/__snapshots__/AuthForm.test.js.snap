// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm component renders AuthForm 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "margin": "auto",
      "width": 400,
    }
  }
>
  <label
    style={
      Object {
        "margin": 10,
      }
    }
  >
    Username:
    <input
      name="username"
      onChange={[Function]}
      type="test"
    />
  </label>
  <label
    style={
      Object {
        "margin": 10,
      }
    }
  >
    Password:
    <input
      name="password"
      onChange={[Function]}
      type="password"
    />
  </label>
  <button>
    Submit
  </button>
  <span>
    error
  </span>
</form>
`;
